test/unit/math/prim/mat/prob/ordered_probit_test.cpp

```diff
@@ -0,0 +1,202 @@
+    EXPECT_FLOAT_EQ(log(theta(k)),ordered_probit_log(k+1,lambda,c));
+
+  EXPECT_THROW(ordered_probit_log(0,lambda,c),std::domain_error);
+  EXPECT_THROW(ordered_probit_log(4,lambda,c),std::domain_error);
+}
+
+TEST(ProbDistributions,ordered_probit) {
+  using stan::math::ordered_probit_log;
+  int K = 4;
+  Eigen::Matrix<double,Eigen::Dynamic,1> c(K-1);
+  c << -0.3, 0.1, 1.2;
+  double lambda = 0.5;
+  EXPECT_THROW(ordered_probit_log(-1,lambda,c),std::domain_error);
+  EXPECT_THROW(ordered_probit_log(0,lambda,c),std::domain_error);
+  EXPECT_THROW(ordered_probit_log(5,lambda,c),std::domain_error);
+  for (int k = 1; k <= K; ++k)
+    EXPECT_NO_THROW(ordered_probit_log(k,lambda,c));
+
+  Eigen::Matrix<double,Eigen::Dynamic,1> c_zero; // init size zero
+  EXPECT_EQ(0,c_zero.size());
+  EXPECT_THROW(ordered_probit_log(1,lambda,c_zero),std::domain_error);
+
+  Eigen::Matrix<double,Eigen::Dynamic,1> c_neg(1);
+  c_neg << -13.7;
+  EXPECT_NO_THROW(ordered_probit_log(1,lambda,c_neg));
+
+  Eigen::Matrix<double,Eigen::Dynamic,1> c_unord(3);
+  c_unord << 1.0, 0.4, 2.0;
+  EXPECT_THROW(ordered_probit_log(1,lambda,c_unord),std::domain_error);
+
+  Eigen::Matrix<double,Eigen::Dynamic,1> c_unord_2(3); 
+  c_unord_2 << 1.0, 2.0, 0.4;
+  EXPECT_THROW(ordered_probit_log(1,lambda,c_unord_2),std::domain_error);
+
+  double nan = std::numeric_limits<double>::quiet_NaN();
+  double inf = std::numeric_limits<double>::infinity();
+
+  EXPECT_THROW(ordered_probit_log(1,nan,c),std::domain_error);
+  EXPECT_THROW(ordered_probit_log(1,inf,c),std::domain_error);
+
+  Eigen::Matrix<double,Eigen::Dynamic,1> cbad(2); 
+  cbad << 0.2, inf;
+  EXPECT_THROW(ordered_probit_log(1,1.0,cbad),std::domain_error);
+  cbad[1] = nan;
+  EXPECT_THROW(ordered_probit_log(1,1.0,cbad),std::domain_error);
+
+  Eigen::Matrix<double,Eigen::Dynamic,1> cbad1(1); 
+  cbad1 <<  inf;
+  EXPECT_THROW(ordered_probit_log(1,1.0,cbad1),std::domain_error);
+  cbad1[0] = nan;
+  EXPECT_THROW(ordered_probit_log(1,1.0,cbad1),std::domain_error);
+
+  Eigen::Matrix<double,Eigen::Dynamic,1> cbad3(3); 
+  cbad3 <<  0.5, inf, 1.0;
+  EXPECT_THROW(ordered_probit_log(1,1.0,cbad3),std::domain_error);
+  cbad3[1] = nan;
+  EXPECT_THROW(ordered_probit_log(1,1.0,cbad3),std::domain_error);
+}
+
+void expect_nan(double x) {
+  EXPECT_TRUE(std::isnan(x));
+}
+
+
+TEST(ProbDistributionOrderedProbit, error_check) {
+  boost::random::mt19937 rng;
+  double inf = std::numeric_limits<double>::infinity();
+  Eigen::VectorXd c(4);
+  c << -2, 
+    2.0,
+    5,
+    10;
+  EXPECT_NO_THROW(stan::math::ordered_probit_rng(4.0, c, rng));
+
+  EXPECT_THROW(stan::math::ordered_probit_rng(stan::math::positive_infinity(), 
+                                                c, rng),
+               std::domain_error);
+  c << -inf, 
+    2.0,
+    -5,
+    inf;
+  EXPECT_THROW(stan::math::ordered_probit_rng(4.0, c, rng),
+               std::domain_error);
+
+}
+
+TEST(ProbDistributionOrderedProbit, chiSquareGoodnessFitTest) {
```

Comment:
Great!  I'm glad you could figure this all out.  @bbbales2 is going through and vectorizing the RNGs, and this is what he'll need to do this one when he gets to them.